sdk/privatedns/Azure.ResourceManager.PrivateDns/tests/Scenario/RecordTests.cs

```diff
@@ -0,0 +1,482 @@
+            Assert.IsNotEmpty(list);
+            ValidateRecordBaseInfo(list.First(item => item.Data.Name == ptrRecordName).Data, ptrRecordName);
+
+            // Delete
+            await ptrRecord.Value.DeleteAsync(WaitUntil.Completed);
+            flag = await collection.ExistsAsync(ptrRecordName);
+            Assert.IsFalse(flag);
+        }
+
+        [RecordedTest]
+        public async Task SoaRecordOperationTest()
+        {
+            var collection = _privateDns.GetPrivateDnsSoaRecords();
+            string soaRecordName = "@";
+
+            // CreateOrUpdate
+            var data = new PrivateDnsSoaRecordData()
+            {
+                TtlInSeconds = 3600
+            };
+            var soaRecord = await collection.CreateOrUpdateAsync(WaitUntil.Completed, soaRecordName, data);
+            ValidateRecordBaseInfo(soaRecord.Value.Data, soaRecordName);
+            Assert.IsNotNull(soaRecord.Value.Data.PrivateDnsSoaRecord.Email);
+            Assert.AreEqual("privateDnsZones/SOA", soaRecord.Value.Data.ResourceType.Type.ToString());
+            Assert.AreEqual(3600, soaRecord.Value.Data.TtlInSeconds);
+            Assert.IsTrue(soaRecord.Value.Data.PrivateDnsSoaRecord.RefreshTimeInSeconds > 0);
+            Assert.IsTrue(soaRecord.Value.Data.PrivateDnsSoaRecord.RetryTimeInSeconds > 0);
+            Assert.IsTrue(soaRecord.Value.Data.PrivateDnsSoaRecord.ExpireTimeInSeconds > 0);
+            Assert.IsTrue(soaRecord.Value.Data.PrivateDnsSoaRecord.MinimumTtlInSeconds > 0);
+
+            // Exist
+            bool flag = await collection.ExistsAsync(soaRecordName);
+            Assert.IsTrue(flag);
+
+            // Update
+            var updateResponse = await soaRecord.Value.UpdateAsync(new PrivateDnsSoaRecordData() { TtlInSeconds = 7200 });
+            Assert.AreEqual(7200, updateResponse.Value.Data.TtlInSeconds);
+
+            // Get
+            var getResponse = await collection.GetAsync(soaRecordName);
+            ValidateRecordBaseInfo(getResponse.Value.Data, soaRecordName);
+            Assert.IsNotNull(getResponse.Value.Data.PrivateDnsSoaRecord.Email);
+            Assert.AreEqual("privateDnsZones/SOA", getResponse.Value.Data.ResourceType.Type.ToString());
+            Assert.AreEqual(7200, getResponse.Value.Data.TtlInSeconds);
+            Assert.IsTrue(getResponse.Value.Data.PrivateDnsSoaRecord.RefreshTimeInSeconds > 0);
+            Assert.IsTrue(getResponse.Value.Data.PrivateDnsSoaRecord.RetryTimeInSeconds > 0);
+            Assert.IsTrue(getResponse.Value.Data.PrivateDnsSoaRecord.ExpireTimeInSeconds > 0);
+            Assert.IsTrue(getResponse.Value.Data.PrivateDnsSoaRecord.MinimumTtlInSeconds > 0);
+
+            // GetAll
+            var list = await collection.GetAllAsync().ToEnumerableAsync();
+            Assert.IsNotEmpty(list);
+            ValidateRecordBaseInfo(list.First(item => item.Data.Name == soaRecordName).Data, soaRecordName);
+
+            // Delete - Type SOA cannot delete
```

Comment:
Do you know what will happen if we call delete method for SOA? If the service will return 400 or some other errors, how about we just remove this method from `PrivateDnsSoaRecordResource`?

Reply:
It will return `400(Bad Request){"code":"BadRequest","message":"Record sets of type 'SOA' with name '@' cannot be deleted."}`
I remove that method in customization, please double confirm. If we confirm that change, I will remove the delete methods in dns.